registration/registration-services/src/main/java/io/mosip/registration/mdm/service/impl/MosipDeviceSpecificationHelper.java

```diff
@@ -278,18 +278,22 @@ public String getHttpClientResponseEntity(String url, String method, String body
 	public CloseableHttpResponse getHttpClientResponse(String url, String method, String body) throws IOException {
 		int timeout = getMDMConnectionTimeout(method);
 		LOGGER.debug("MDM HTTP CALL method : {}  with timeout {}", method, timeout);
+		try(CloseableHttpClient client = HttpClients.createDefault()){
 		RequestConfig requestConfig = RequestConfig.custom()
 				.setConnectTimeout(timeout)
 				.setSocketTimeout(timeout)
 				.setConnectionRequestTimeout(timeout)
 				.build();
-		CloseableHttpClient client = HttpClients.createDefault();
 		StringEntity requestEntity = new StringEntity(body, ContentType.create("Content-Type", Consts.UTF_8));
 		HttpUriRequest httpUriRequest = RequestBuilder.create(method)
 				.setConfig(requestConfig)
 				.setUri(url)
 				.setEntity(requestEntity)
 				.build();
 		return client.execute(httpUriRequest);
+		} catch (Exception exception) {
+			LOGGER.error(ExceptionUtils.getStackTrace(exception));
+			return null;
```

Comment:
Are we handling null in the caller method? And for the logger this way is better LOGGER.error("MDM call failed", ex);